dockerfiles/stages/3-opam-deps

```diff
@@ -0,0 +1,48 @@
+#################################################################################################
+# The "opam-deps" Stage
```

Comment:
what's the workflow when I'm a dev who wants to change our opam deps? do I rebuild this and push it?

Reply:
```
Remove opam.tar.gz caching and instead include all build-time dependencies in MinaToolchain
dockerfiles/stages/2-toolchain
dockerfiles/stages/3-opam-deps
buildkite/src/Command/RunInToolchain.dhall
buildkite/src/Command/OpamInit.dhall (deleted)
Still some work remaining to automatically rebuild this when there are new dependencies. CI will build it, but you still have to manually update buildkite/src/Constants/ContainerImages.dhall
```

^^ cool CI does it for you 💯